msal/src/Microsoft.Identity.Client/CacheV2/TokenCacheV2.cs

```diff
@@ -0,0 +1,234 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Microsoft.Identity.Client.CacheV2.Impl;
+using Microsoft.Identity.Core.Cache;
+
+namespace Microsoft.Identity.Client.CacheV2
+{
+    /// <summary>
+    /// In the V2 cache, the ITokenCache interface will provide the existing infra used by external developers
+    /// to serialize/deserialize if they're managing their own cache.
+    /// But the cache implementation does not live within the TokenCacheV2 infra, it is simply a facade
+    /// into the StorageManager and other pieces of the cache that are owned by the ClientApplicationBase.
+    /// </summary>
+    internal class TokenCacheV2 : ITokenCache
+    {
+        private readonly object _lock = new object();
+
+        //private TokenCacheNotification _afterAccess;
+        //private TokenCacheNotification _beforeAccess;
+        //private TokenCacheNotification _beforeWrite;
+        private CacheData _deserializedCacheDataHolding = new CacheData();
+        private IStorageManager _storageManager;
```

Comment:
not sure I understand the scenario for either having a storage manager or not... why is the facade doing 2 things_